clients/client-java/src/main/java/org/apache/gravitino/client/KerberosTokenProvider.java

```diff
@@ -155,6 +135,68 @@ void setHost(String host) {
     this.host = host;
   }
 
+  private interface SubjectProvider {
+    Subject get() throws LoginException;
+
+    void close() throws LoginException;
+  }
+
```

Comment:
Could you explain the scenarios of `ExistingSubjectProvider` and `LoginSubjectProvider`? and any reason to add this to Flink authentication PR?

Reply:
The Flink has logined using keytab and principal. We should reuse the credential created by Flink. If the framework maintain the credential. The client won't need to maintain the credential. We don't need to login again. 
ExistingSubjectProvider is used for reusing the framework's credential.
LoginSubjectProvider will use keytab and principal to login, generate credentail.